mkdocs/docs/HPC/setting_up_python_virtual_environments.md

```diff
@@ -0,0 +1,221 @@
+# Python Virtual Environments (venv's)
+
+## Introduction
+
+Python virtual environments (venv's) are a way to create isolated Python environments.
+You can install packages in a venv without affecting the system Python installation.
+Because a normal user cannot install packages globally, 
+venv's are a way to install packages without needing root access.
+This is useful when you need to use a package that is not available as a module on the HPC cluster.
+
+## Creating a venv
+
+A venv can be created with the following commands:
+
+```bash
+$ python -m venv myenv      # Create a new venv named myenv
+$ source myenv/bin/activate # Activate the venv
+```
+
+After activating the venv, you can install packages with `pip`:
+
+```bash
+$ pip install example_package1
+$ pip install example_package2
+```
+
+It is now possible to run Python scripts that use the installed packages in the venv. To deactivate the venv, run:
+
+```bash
+$ deactivate
+```
+
+!!! note
+    Always prefer to use modules if they are available. 
+    Modules are compiled and optimized for the HPC cluster, while packages installed with `pip` are not.
+    To check if a package is available as a module, you can use the following command:
+
+    ```bash
+    $ module av package_name
```

Comment:
We should also mention that some Python packages are installed as *extensions*, which are part of a particular environment module. Classic examples are `numpy`, `scipy`, `pandas`, which are part of the `SciPy-bundle` module.

You can use `module show` on a `SciPy-bundle` module to check which extensions it provides.

(With some work, we can actually also make extensions visible via `module avail`, but that's not trivial, we haven't found time for it yet. @lbarraga Maybe that's something you can help us with in the coming weeks...)

Reply:
Added info about bundle modules in [b80340f](https://github.com/hpcugent/vsc_user_docs/pull/639/commits/b80340f01448eb04d010a44d05285b28fece3cc4)